### DISCUSSION TITLE
Shelly EM - no energy report

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16546] [Q&A] [answered]
After converting a Shelly em to tasmota, all seems to be working ok but no energy information.
I doubt it's a bad device since it was working before changing it to tasmota.
In fact it is still working, just no power monitoring.

I found this and it seems the template and everything is good so I guess it means I need a more custom firmware for this one.
https://github.com/arendst/Tasmota/issues/14261

Everything looks right on the device.

```
backlog template
22:09:41.397 CMD: backlog template
22:09:41.429 RSL: RESULT = {"NAME":"Shelly EM","GPIO":[0,0,0,0,0,0,0,0,640,3456,608,224,0,1],"FLAG":0,"BASE":18}```

module
22:06:22.952 CMD: module
22:06:22.956 RSL: RESULT = {"Module":{"0":"Shelly EM"}}

gpio
22:07:19.257 CMD: gpio
22:07:19.261 RSL: RESULT = {"GPIO17":{"0":"None"}}

i2cscan
22:08:01.641 CMD: i2cscan
22:08:01.661 RSL: RESULT = {"I2CScan":"Device(s) found at 0x38 0x51"}

i2cdriver
22:09:19.699 CMD: i2cdriver
22:09:19.704 RSL: RESULT = {"I2CDriver":"7"}
```

I used the Shelly EM build
https://github.com/yaourdt/mgos-to-tasmota

Then I upgraded the firmware with my own build since it has TLS and the above one didn't and is minimal.

>May be retry a serial reflash of official Tasmota with full erase.
>Configure the device without mqtt to get proper power values on web page.
>Once this is ok, upgrade to your firmware and check that power values are still here.
>Move on configuring mqtt only if those steps are ok

I have done all of the above.
No MQTT, using the default Firmware upgrade from tasmota with template but still shows no power values.

Program Version | 12.1.1(tasmota)
Build date /time 2022-08-25T11:33:55
Core/SDK version 2_7_4_9/2.2.2-dev(38a443e)
Flash write count 48 at 0xFA000
Boot count 19

ESP Chip Id | 16572811 (ESP8266EX)
Flash chip ID 0x1620C2
Flash size 4096 KB
Program flash size 1024 KB
Program size 624 KB
Free program space 376 KB
Free memory 25.7 KB

>use official tasmota.bin.gz
>command : reset 2 will erase all configs to insure restarting from scratch, including wifi
>configure minimum : wifi, template (no mqtt)
>power cycle de device (disconnect from mains, wait 30 sec, reconnect)

I followed the steps exactly and still no power readings.

Anyone else running this Shelly that might have ideas, suggestions? I don't have anything small enough to get into serial mode and can't seem to find a ready made cable to buy.

**Comment by barbudor** [comment]
Hi @isabido, could you help @dac2020?
Thanks

**Reply by isabido** [reply]
Sorry for the delay! I can think of 3 possible options.

1º After flashing tasmota for the first time, you have to remove power again. It's not just a soft reset.

2º Load this profile

{"NAME":"Shelly EM","GPIO":[0,0,0,0,0,0,0,0,640,3457,608,224,0,1],"FLAG":0,"BASE": 18}

3º Then you have to configure this option.

![image](https://user-images.githubusercontent.com/559650/190862148-d1b408a1-e44c-4b7e-bc80-776415df3965.png)

4º Finally, activate this option from the console, so that you have a measure regardless of how the TOGGLE is.

SetOption21 on

It should work after all these steps. And tell me.

**Reply by alexdelprete** [reply]
> 2º Load this profile
> 
> {"NAME":"Shelly EM","GPIO":[0,0,0,0,0,0,0,0,640,3457,608,224,0,1],"FLAG":0,"BASE": 18}

Hi isabido,

what's the difference with the template here: https://templates.blakadder.com/shelly_EM.html

I've used that one on 3 Shelly EMs, 2 of them are working fine, but one of them is having boot loop issues (it resets every couple of days) and I've also used `SetOption65 1` as a precaution, even though I didn't think power was the issue.

So digging for possible solutions, I noticed the template I used is different vs the one recommended by some people in various issues.

Thanks on any tip on this problem.

**Reply by sfromis** [reply]
The one suggested was at the time the one suggested in the templates repository. The current one there (as of 2022-10-22) has been amended to also include gpio 16 configured as "ADE7953 RST" to more fully represent the hardware.

**Reply by alexdelprete** [reply]
> The one suggested was at the time the one suggested in the templates repository. The current one there (as of 2022-10-22) has been amended to also include gpio 16 configured as "ADE7953 RST" to more fully represent the hardware.

Thank you very much for the explanation.

**Comment by dac2020** [comment, marked as answer]
Hi isabido,

Yes, those were the steps but there was one crucial one missing.

I went ahead and re-did as suggested.
Re-flashed a minimal, then my own build which has TLS. 
Turned power off for 30 seconds and restarted it.
Checked option you show, it was already there.
"SetOption21 on" was never done since I never came across that anywhere while searching.

![image](https://user-images.githubusercontent.com/9727841/190863217-6f82d97c-b9fe-4034-b78f-fece0e7d53fc.png)

Have to move my clamps around now to get proper readings but all seems to be working.

That did the trick. Thank you!
